GeeksCoreLibrary/Modules/HealthChecks/Services/DatabaseHealthService.cs

```diff
@@ -1,65 +1,116 @@
     : IHealthCheck
 {
     private readonly GclSettings gclSettings = gclSettings.Value;
 
-    /// <inheritdoc />
-    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context,
-        CancellationToken cancellationToken = new CancellationToken())
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
     {
-        // A new query to check the max active connections from the DatabaseHealthCheck.
-        var query = "SELECT COUNT(*) AS active_connections FROM information_schema.PROCESSLIST;";
-        var datatable = await databaseConnection.GetAsync(query);
-
-        var healthCheckConnections = gclSettings.HealthChecks.MaximumDatabaseConnections;
-        var healthCheckConnectionsTime = gclSettings.HealthChecks.MaximumConnectionsInTime;
+        var messages = new List<string>(); 
+        var isHealthy = true; 
 
-        // If no value is set, we are skipping this test.
-        if (healthCheckConnections > 0)
+        try
         {
-            // Retrieve the count of active connections.
-            var activeConnections = Convert.ToInt32(datatable.Rows[0]["active_connections"]);
+            // ✅ Check if database service is available
+            if (databaseConnection == null)
+            {
+                return HealthCheckResult.Unhealthy("Database connection service is not available.");
+            }
 
-            // Check if the number of active connections exceeds the limit.
-            if (activeConnections > healthCheckConnections)
+            // ✅ Ensure health check is enabled
+            if (gclSettings.HealthChecks?.DatabaseHealthCheckEnabled != true)
             {
-                return HealthCheckResult.Unhealthy($"Too many database connections. There are currently {activeConnections} connections active, which is more than the threshold {healthCheckConnections} that is set in the appSettings.");
+                return HealthCheckResult.Unhealthy("Database health check settings are missing or misconfigured.");
             }
-        }
 
-        // If no value is set, we are skipping this test.
-        if (healthCheckConnectionsTime > 0)
-        {
-            // Query to check the max open connections in time from the DatabaseHealthCheck.
-            query = "SELECT ID, USER, HOST, TIME AS connection_time_in_sec FROM information_schema.PROCESSLIST WHERE USER != 'repluser' AND HOST != 'localhost' ORDER BY connection_time_in_sec DESC";
-            datatable = await databaseConnection.GetAsync(query);
-            if (datatable.Rows.Count == 0)
+            //  Basic Database Connectivity Check
+            var connectionTestQuery = "SELECT 1;";
+            var connectionTestTable = await databaseConnection.GetAsync(connectionTestQuery);
+            if (connectionTestTable.Rows.Count == 0)
+            {
```

Comment:
This is a nice idea, but it will never work this way. If the database is not reachable, then you will get an exception. If it is reachable, then you will always get a result with that query, so the code inside this if statement will never be reached.

Instead of this if statement, you could add a catch for `MySqlException`, those exceptions will always be thrown when a query failed for any reason. Those exceptions have a property called `ErrorCode`, you can/should use that property to see what message you can return in your output